lib/api.js

```diff
@@ -74,47 +74,73 @@ api.declare({
       ClientToken,
       Region,
       SpotPrice,
-      LaunchSpecification,
+      RequestType,
+      LaunchInfo,
     } = req.body;
 
-    // This is a crtical check to ensure that we only ever try to create spot requests
-    // which are using the correct key pair name
+    // This is a crtical check to ensure that we only ever try to create
+    // instances which are using the correct key pair name
     let keyName = createKeyPairName(this.keyPrefix, this.instancePubKey, workerType);
-    if (keyName !== LaunchSpecification.KeyName) {
+    if (keyName !== LaunchInfo.KeyName) {
       log.error({
         requiredKeyName: keyName,
-        providedKeyName: LaunchSpecification.KeyName,
+        providedKeyName: LaunchInfo.KeyName,
       }, 'KeyName requirement not met');
-      return res.reportError('InputError', 'LaunchSpecification is invalid!');
+      return res.reportError('InputError', 'LaunchInfo is invalid!');
     }
 
-    assert(keyName === LaunchSpecification.KeyName);
+    assert(keyName === LaunchInfo.KeyName);
+
+    if (RequestType === 'spot') {
+      LaunchInfo.InstanceMarketOptions = {
+        MarketType: 'spot',
+        SpotOptions: {
+          SpotInstanceType: 'one-time',
+        },
+      };
+
+      // using typeof to account for a 0.0 value... I guess no one would use a
```

Comment:
I think I am losing some context, but I didn't understand how the `typeof` handles the 0.0 case...

Reply:
```js
$ node
> if (0.0) { console.log('truthy') }
undefined
> if (typeof 0.0 === 'number') { console.log('truthy') }
truthy
undefined
> 
```
It's being pedantic, but I want to make sure that if a value of 0 is passed, which is a falsy value, it still triggers this value to be added to the request.  I know a bid of 0 is not a great thing, but I'm worried about the case where some automated process starts accidentally calculating bids of 0$.